src/coreclr/tools/Common/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -333,6 +335,47 @@ private static unsafe int GetLoongArch64PC12(uint* pCode)
             return imm;
         }
 
+        private static unsafe int GetRiscv64PC12(uint* pCode)
+        {
+            uint auipcInstr = *pCode;
+
+            // first get the high 20 bits,
+            int imm = (int)(((auipcInstr >> 12) & 0xFFFFF) << 12);
+
+            // then get the low 12 bits,
+            auipcInstr = *(pCode + 1);
+            imm += ((int)(((auipcInstr >> 20) & 0xFFF) << 20)) >> 20;
+
+            return imm;
+        }
+
+        private static unsafe void PutRiscv64PC12(uint* pCode, long imm32)
+        {
+            // Verify that we got a valid offset
+            Debug.Assert((int)imm32 == imm32);
+
+            uint auipcInstr = *pCode;
+            Debug.Assert((auipcInstr & 0x1F) == 0x00000017);  // Must be auipc
+
+            int relOff = (int)imm32 & 0x800;
+            int imm = (int)imm32 + relOff;
+            relOff = ((imm & 0x7ff) - relOff) & 0xfff;
```

Comment:
Isn't it okay to make codes similar to codes in `jit/emitriscv64.cpp`?
https://github.com/dotnet/runtime/blob/cc4db9bd71f2dfcc29dc71385cd96916253c8850/src/coreclr/jit/emitriscv64.cpp#L2423-L2433

Reply:
Done